### PR TITLE
Bugfix/180 entitylinking improvements

### PR DESCRIPTION
**Pull request by rcffc**

### CODE REVIEW

File: inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/ConceptLinkingService.java [hunk 1]
```diff
@@ -149,24 +146,26 @@ private synchronized Sentence getMentionSentence(JCas aJcas, int aBegin)
         Set<CandidateEntity> candidates = new HashSet<>();
         List<String> mentionArray = Arrays.asList(aMention.split(" "));
```
Comment (Rentier) [inline inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/ConceptLinkingService.java]: You name a list array

File: inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/ConceptLinkingService.java [hunk 2]
```diff
@@ -149,24 +146,26 @@ private synchronized Sentence getMentionSentence(JCas aJcas, int aBegin)
...
         if (stopwords != null) {
-            if (mentionArray.stream().allMatch(m -> stopwords.contains(m))) {
+            if (stopwords.containsAll(mentionArray)) {
                 logger.error("Mention [{}] consists of stopwords only - returning.", aMention);
                 return Collections.emptySet();
             }
         }
 
-        if (mentionArray.isEmpty()) {
+        aMention = String.join(" ", mentionArray);
```
Comment (Rentier) [inline inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/ConceptLinkingService.java]: Do not reassign function arguments, e.g. the ones prefixed with `a`.

File: inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/recommender/NamedEntityLinker.java [hunk 3]
```diff
@@ -131,7 +131,7 @@ public void setProject(Project project)
     {
         List<List<List<AnnotationObject>>> result = new ArrayList<>();
 
-        for (List<T> sentence : inputData) {
+        inputData.parallelStream().forEach(sentence -> {
```
Comment (Rentier) [inline inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/recommender/NamedEntityLinker.java]: Did you measure whether it is actually faster?

Comment (rcffc) [reply inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/recommender/NamedEntityLinker.java]: Yes, it was twice as fast. Evaluated with 16 sentences annotated with NEs.
[NER-de-sample.tsv.conll.zip](https://github.com/inception-project/inception/files/2000127/NER-de-sample.tsv.conll.zip)

**Comment by reckart** [comment]
@rcffc I would like to merge the branch for the 0.3.1 release, but it has conflicts. Can you fix it with minimal effort or should the changes be moved to 0.3.2/0.4.0?

**Pull request merged**